ferrocene/tools/self-test/src/compile.rs

```diff
@@ -119,12 +135,35 @@ fn compile(ctx: &Context<'_>, program: &SampleProgram) -> Result<(), Error> {
     }
     cmd.args(program.rustflags);
     cmd.args(&ctx.target.rustflags);
-    cmd.arg(program_path);
+    cmd.arg(&program_path);
 
     run_command(&mut cmd).map_err(|error| Error::SampleProgramCompilationFailed {
         name: program.name.into(),
         error,
     })?;
+
+    if let Some(expected_output) = expected_output {
+        // where is it
+        let bin_name = program.name.replace(".rs", "");
+        let bin_path = ctx.output_dir.join(bin_name);
+        // now try and execute it
+        let mut cmd = Command::new(&bin_path);
+        let output = cmd.output().map_err(|error|
+            // TODO change for RunningSampleProgramFailed
+            Error::RunningSampleProgramFailed {
+                name: program.name.into(),
+                error,
+            })?;
+        if output.stdout != expected_output {
+            // TODO change for SampleProgramOutputWrong
+            return Err(Error::SampleProgramOutputWrong {
+                name: program.name.into(),
+                expected: expected_output.to_vec(),
+                found: output.stdout,
+            });
```

Comment:
Why the TODOs here?

Reply:
ah, because I fixed the TODO and didn't delete the note I'd left myself